internal/domain/file_content_response.go

```diff
@@ -0,0 +1,9 @@
+package domain
+
+type FileContentResponse struct {
+	Filename  string `json:"filename"`
```

Comment:
Consider adding validation logic to ensure that 'StartLine' is less than or equal to 'EndLine'. This could prevent logical errors when handling file content segments.